lib/tools/reducers/master.js

```diff
@@ -0,0 +1,23 @@
+import { getPostfixFile } from '../files.js';
+import XMLFilterWriter from '../../xml/XMLFilterWriter.js';
+
+/**
+ * @param {import('#types').XMLTagFilter} productFilter
+ * @param {Array<string>} files
+ */
+export default async (productFilter, files) => Promise.all(files.map(async (masterFile) => {
+    const outFile = getPostfixFile(masterFile);
+
+    const tempFileWithFilteredProducts = outFile + '.temp';
+
+    const masterFilterByProduct = new XMLFilterWriter(masterFile, tempFileWithFilteredProducts);
+
+    /**
+     * Filter product by final registry in navigation catalog
+     */
+    await masterFilterByProduct.startAsync('product', productFilter);
+
+    const masterFilterByAssignments = new XMLFilterWriter(tempFileWithFilteredProducts, outFile);
+
+    return masterFilterByAssignments.startAsync('category-assignment', productFilter);
```

Comment:
it not return any value from promise, so here better to write "await" instead of "return"